doc/man7/provider-signature.pod

```diff
@@ -0,0 +1,229 @@
+ OP_signature_sign                   OSSL_FUNC_SIGNATURE_SIGN
+
+ OP_signature_verify_init            OSSL_FUNC_SIGNATURE_VERIFY_INIT
+ OP_signature_verify                 OSSL_FUNC_SIGNATURE_VERIFY
+
+ OP_signature_verify_recover_init    OSSL_FUNC_SIGNATURE_VERIFY_RECOVER_INIT
+ OP_signature_verify_recover         OSSL_FUNC_SIGNATURE_VERIFY_RECOVER
+
+ OP_signature_set_params             OSSL_FUNC_SIGNATURE_SET_PARAMS
+
+A signature algorithm implementation may not implement all of these functions.
+In order to be a consistent set of functions a provider must implement
+OP_signature_newctx and OP_signature_freectx.
+It must also implement both of OP_signature_sign_init and OP_signature_sign,
+or both of OP_signature_verify_init and OP_signature_verify, or both of
+OP_signature_verify_recover_init and OP_signature_verify_recover.
+All other functions are optional.
+
+A signature algorithm must also implement some mechanism for generating,
+loading or importing keys via the key management (OSSL_OP_KEYMGMT) operation.
+See L<provider-keymgmt(7)> for further details.
+
+=head2 Context Management Functions
+
+OP_signature_newctx() should create and return a pointer to a provider side
+structure for holding context information during a signature operation.
+A pointer to this context will be passed back in a number of the other signature
+operation function calls.
+The paramater B<provctx> is the provider context generated during provider
+initialisation (see L<provider(3)>).
+
+OP_signature_freectx() is passed a pointer to the provider side signature
+context in the B<ctx> parameter.
+This function should free any resources associated with that context.
+
+OP_signature_dupctx() should duplicate the provider side signature context in
+the B<ctx> parameter and return the duplicate copy.
+
+=head2 Signing Functions
+
+OP_signature_sign_init() initialises a context for signing given a provider side
+signature context in the B<ctx> paramter, and a pointer to a provider key object
+in the B<provkey> parameter.
+The key object should have been previously generated, loaded or imported into
+the provider using the key management (OSSL_OP_KEYMGMT) operation (see
+provider-keymgmt(7)>.
+
+OP_signature_sign() performs the actual signing itself.
+A previously initialised signature context is passed in the B<ctx>
+parameter.
+The data to be signed is pointed to be the B<tbs> parameter which is B<tbslen>
+bytes long.
+Unless B<sig> is NULL, the signature should be written to the location pointed
+to by the B<sig> parameter and it should not exceed B<sigsize> bytes in length.
+The length of the signature should be written to B<*siglen>.
+If B<sig> is NULL then the maximum length of the signature should be written to
+B<*siglen>.
+
+=head2 Verify Functions
+
+OP_signature_verify_init() initialises a context for verifying a signature given
+a provider side signature context in the B<ctx> paramter, and a pointer to a
```

Comment:
param

Reply:
_parameter_

Reply:
I think there's something wrong with my keyboard. It doesn't seem to be able to write parameter correctly.

Reply:
Oh right, blame the keyboard :-D